app/os_android.go

```diff
@@ -776,6 +780,13 @@ func (w *window) setVisible(env *C.JNIEnv) {
 	if width == 0 || height == 0 {
 		return
 	}
+	displayModeCount := int(C.jni_CallIntMethod(env, w.view, gioView.countDisplayModes))
+	if displayModeCount > 1 {
+		// Only do the main thread work on devices that need it
+		runOnMainWithEnv(env, func(mainEnv *C.JNIEnv) {
+			callVoidMethod(mainEnv, w.view, gioView.setHighRefreshRate)
+		})
+	}
```

Comment:
I'm surprised you need this for every `setVisible`. Isn't it enough to set the display mode once, in the `GioView` Java constructor?

Reply:
Looks like it is, much more straightforward this way.